### ISSUE TITLE
[RF] HistFactory workspaces incompatible between versions 6.26 and 6.28

### ISSUE THREAD

**Issue by alexander-held** [root-project/root#12967]
### Check duplicate issues.

- [X] Checked for duplicates

### Describe the bug

It is not possible to operate on a HistFactory Measurement object in a workspace created with ROOT 6.26 by using ROOT 6.28. Attempting to do so results in the following:
```
Error in <TRint::HandleTermInput()>: std::length_error caught: basic_string::_M_replace_aux
```

The opposite scenario (workspace created in 6.28, using 6.26 to read it) also causes issues. This means that there is essentially some trial&error when handling unknown workspaces to find a suitable ROOT version.

### What is the expected behaviour?

Unclear to me whether this breaking change is expected or not, but the error message is not particularly clear in case this is expected.

### How to reproduce?

Create HistFactory workspace in ROOT 6.26 (or use this example https://cernbox.cern.ch/s/uRaTglosZypwm81, created from xmls via `hist2workspace` using 6.26/08), here called `workspace_example.root`:

```bash
root@2cd219ab253b:~/data# root --version
ROOT Version: 6.28/04
Built for linuxx8664gcc on May 24 2023, 17:37:00
From tags/v6-28-04@v6-28-04
root@2cd219ab253b:~/data# root -l workspace_example.root
root [0]
Attaching file workspace_example.root as _file0...
(TFile *) 0x402fde6770
root [1] minimal_example->PrintXML()
Error in <TRint::HandleTermInput()>: std::length_error caught: basic_string::_M_replace_aux
```

### ROOT version

6.28/04

### How did you install ROOT?

custom build, container via https://gitlab.cern.ch/atlas-amglab/atlstats

### Which operating system are you using?

N/A

**Comment by guitargeek** [comment]
Okay I found the fix: adding some missing entries to the `LinkDef.h`. This will be fixed in the next 6.28 patch release, which is 6.28.06.

**Comment by alexander-held** [comment]
Thanks a lot for looking into this! Great to hear that this will be fixed in the next release.
